strict-checked-vars/test/Test/Control/Concurrent/Class/MonadMVar/Strict/Checked/WHNF.hs

```diff
@@ -0,0 +1,312 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Test.Control.Concurrent.Class.MonadMVar.Strict.Checked.WHNF where
+
+import           Control.Concurrent.Class.MonadMVar.Strict.Checked hiding
+                     (newEmptyMVar, newEmptyMVarWithInvariant, newMVar,
+                     newMVarWithInvariant)
+import qualified Control.Concurrent.Class.MonadMVar.Strict.Checked as Checked
+import           Control.Monad (void)
+import           Control.Monad.IOSim (runSimOrThrow)
+import           Data.Typeable (Typeable)
+import           NoThunks.Class (OnlyCheckWhnf (..), unsafeNoThunks)
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.QuickCheck (Fun, Property, applyFun, counterexample,
+                     ioProperty, property, testProperty, (.&&.))
+import           Test.Utils (Invariant (..), noInvariant, trivialInvariant,
+                     whnfInvariant, (..:))
+
+{-------------------------------------------------------------------------------
+  Main test tree
+-------------------------------------------------------------------------------}
+
+tests :: TestTree
+tests = testGroup "WHNF" [
+      testGroup "IO" [
+          testIO    "No invariant"      noInvariant
```

Comment:
Why not provide a `QuickCheck` instance for `Invariant`?

Reply:
Done